File: src/main/java/co/herod/adbwrapper/rx/MuteErrorTransformer.java
package co.herod.adbwrapper.rx;

import co.herod.adbwrapper.util.StringUtils;
import io.reactivex.Observable;
import io.reactivex.ObservableSource;
import io.reactivex.ObservableTransformer;

public class MuteErrorTransformer<T> implements ObservableTransformer<T, String> {

    @Override
    public ObservableSource<String> apply(final Observable<T> upstream) {
        return upstream.map(blah -> "").onErrorReturn(a -> "").filter(StringUtils::isNotEmpty);
    }
}

File: src/main/java/co/herod/adbwrapper/rx/MuteErrorTransformer.kt
package co.herod.adbwrapper.rx

import io.reactivex.Observable
import io.reactivex.ObservableSource
import io.reactivex.ObservableTransformer

class MuteErrorTransformer<T> : ObservableTransformer<T, String> {

    override fun apply(upstream: Observable<T>): ObservableSource<String> = upstream
            .map { "" }
            .onErrorReturn { "" }
            .filter({ !it.trim { it <= ' ' }.isEmpty() })
}
